packages/test/src/test-worker-deployment-versioning.ts

```diff
@@ -0,0 +1,399 @@
+/**
+ * Tests worker-deployment-based versioning
+ *
+ * @module
+ */
+import assert from 'assert';
+import { randomUUID } from 'crypto';
+import asyncRetry from 'async-retry';
+import { Client } from '@temporalio/client';
+import { toCanonicalString, WorkerDeploymentVersion } from '@temporalio/common';
+import { temporal } from '@temporalio/proto';
+import { Worker } from './helpers';
+import * as activities from './activities';
+import { makeTestFunction } from './helpers-integration';
+import { unblockSignal, versionQuery } from './workflows';
+
+const test = makeTestFunction({ workflowsPath: __filename });
+
+test('Worker deployment based versioning', async (t) => {
```

Comment:
IIUC, there isn't any actual upgrading going on in this test. This test is checking that workflows with different versioning behaviours can run on workers with different deployment options?